matching/repeat.go

```diff
@@ -16,6 +16,10 @@ func (repeatMatch) Matches(password string) []*match.Match {
 	var matches []*match.Match
 
 	lastIndex := 0
+	indices := map[int]bool{
```

Comment:
While I'm sure this prevents an infinite loop, I'd like to understand better why the index is actually staying the same. Could you make a failing test in repeat_test.go so that we can maybe find a better / the proper fix ?

Reply:
In TestCornerCases there are these two string:
"\x80夤\x1b\xbc\xaf\xa6m",
"77036262\xbd\xbfｿ\xef\xbfｿ\xef3233232".

The first causes the function to get stuck at index 6. The second causes it to cycle through some indices repeatedly.

Totally agree that this isn't the right fix. It does get fuzzing to go a bit further (yet to find any further issues).